### PR TITLE
Provide an await alternative example to fetch

### PR DESCRIPTION
**Pull request by brunoais**
### Description

Provide an await alternative example use to fetch().

### Motivation

In my opinion, using async/await is easier to read and use than using the promise's `then`. That is corroborated by the reason why async/await was made in the first place.

### CODE REVIEW

File: files/en-us/web/api/readablestream/pipeto/index.md [hunk 1]
```diff
@@ -70,6 +70,16 @@ fetch('png-logo.png')
 .then((rs) => rs.pipeTo(new FinalDestinationStream()))
 ```
 
+Same but using await:
+```js
```
Comment (wbamberg) [inline files/en-us/web/api/readablestream/pipeto/index.md:74]:
```suggestion
Same but using await:

```js
```

**Review by wbamberg — COMMENTED** [review]
Thanks for your PR!

I had a few comments and questions.

First, I think use of `await` in our examples should be in an `async` function, or people will get confused when it doesn't work at the top level:

```js
async function theThing() {
  const value = await theOtherThing();
  ...
}

theThing();
```

Second, we should have semicolons at the ends of lines, and all those `let` variables should be `const`.

Finally though I am not sure why we are using `await` with either `response.body` or `rs.pipeTo()`, since they don't seem to return promises.

I'm not very familiar with streams and it is hard to tell since this example isn't runnable, but I had a go with https://mdn.github.io/dom-examples/streams/png-transform-stream/, which is similar code:

```js
  fetch('png-logo.png')
  // Retrieve its body as ReadableStream
  .then(response => response.body)
  // Log each fetched Uint8Array chunk
  .then(rs => logReadableStream('Fetch Response Stream', rs))
  // Transform to a PNG chunk stream
  .then(rs => rs.pipeThrough(new PNGTransformStream()))
  // Log each transformed PNG chunk
  .then(rs => logReadableStream('PNG Chunk Stream', rs))
```

...and that seems to work just fine rewritten as:

```js
async function logPNG() {
  const response = await fetch("png-logo.png");
  let rs = logReadableStream("Fetch Response Stream", response.body);
  rs = await rs.pipeThrough(new PNGTransformStream());
  logReadableStream("PNG Chunk Stream", rs);
}

logPNG();
```

**Comment by brunoais** [comment]
Thank you for the insight. I'd go with:
```
(async () => {
  const response = await fetch("png-logo.png");
  let rs = logReadableStream("Fetch Response Stream", response.body);
  rs = await rs.pipeThrough(new PNGTransformStream());
  logReadableStream("PNG Chunk Stream", rs);
})()
```
instead.
Self-executting anonymous function appears to be a better option with less to think about. What do you think?

**Comment by wbamberg** [comment]
> Thank you for the insight. I'd go with:
> 
> ```
> (async () => {
>   const response = await fetch("png-logo.png");
>   let rs = logReadableStream("Fetch Response Stream", response.body);
>   rs = await rs.pipeThrough(new PNGTransformStream());
>   logReadableStream("PNG Chunk Stream", rs);
> })()
> ```
> 
> instead. Self-executting anonymous function appears to be a better option with less to think about. What do you think?

Yes, I'd be happy with that!

Just to be clear, it would need to be rewritten to align with the other example in the page, rather than the example in https://mdn.github.io/dom-examples/streams/png-transform-stream/.

**Comment by brunoais** [comment]
> Just to be clear, it would need to be rewritten to align with the other example in the page,

Do you mean like this?
```
(async () => {
	// Fetch the original image
	const response = await fetch('png-logo.png')
	// Retrieve its body as ReadableStream
	const body = await response.body
	const rs = body.pipeThrough(new PNGTransformStream())
	rs.pipeTo(new FinalDestinationStream())
})()
```

**Comment by wbamberg** [comment]
> > Just to be clear, it would need to be rewritten to align with the other example in the page,
> 
> Do you mean like this?
> 
> ```
> (async () => {
> 	// Fetch the original image
> 	const response = await fetch('png-logo.png')
> 	// Retrieve its body as ReadableStream
> 	const body = await response.body
> 	const rs = body.pipeThrough(new PNGTransformStream())
> 	rs.pipeTo(new FinalDestinationStream())
> })()
> ```

I think it would need to be something like:

```js
(async () => {
  const response = await fetch("png-logo.png");
  const rs = await response.body.pipeThrough(new PNGTransformStream());
  rs.pipeTo(new FinalDestinationStream());
})();
```

**Comment by brunoais** [comment]
> I think it would need to be something like:
> 
> ```js
> (async () => {
>   const response = await fetch("png-logo.png");
>   const rs = await response.body.pipeThrough(new PNGTransformStream());
>   rs.pipeTo(new FinalDestinationStream());
> })();
> ```

I was thinking... With your insight and a quick test... Maybe that's not correct either!
Maybe the correct way is like this:
```js
(async () => {
  const response = await fetch("png-logo.png");
  response.body
    .pipeThrough(new PNGTransformStream())
    .pipeTo(new FinalDestinationStream());
})();
```

I think I misunderstood that `pipeThrough` returns a `Promise` while actually it just returns a `ReadableStream`.
And actually... That code looks way easier to read than with the `then` for each statement actually run.

I hope once this example is correct and goes through that the page becomes much easier to understand!

**Comment by wbamberg** [comment]
> Maybe the correct way is like this:
> 
> ```js
> (async () => {
>   const response = await fetch("png-logo.png");
>   response.body
>     .pipeThrough(new PNGTransformStream())
>     .pipeTo(new FinalDestinationStream());
> })();
> ```
> 
> I think I misunderstood that `pipeThrough` returns a `Promise` while actually it just returns a `ReadableStream`. And actually... That code looks way easier to read than with the `then` for each statement actually run.

Yes, I think you are right. `pipeThrough` is not async and this looks much more sensible!

File: files/en-us/web/api/readablestream/pipeto/index.md [hunk 2]
```diff
@@ -70,6 +70,19 @@ fetch('png-logo.png')
 .then((rs) => rs.pipeTo(new FinalDestinationStream()))
 ```
 
+Same but using await:
```
Comment (wbamberg) [inline files/en-us/web/api/readablestream/pipeto/index.md:73]:
```suggestion
The same example, but using {{jsxref("Operators/await", "await")}}:
```

**Review by wbamberg — APPROVED** [review]
👍 thank you @brunoais ! I think it would be good to fix the `then` example in this page (and others, probably) and the live example in mdn/dom-examples as well. But I don't want to block this change on that.

**Pull request merged**